frontend/public/components/storage/attach-storage.tsx

```diff
@@ -0,0 +1,468 @@
+import {
+  k8sGet,
+  k8sUpdate,
+  K8sResourceKind,
+  referenceFor,
+} from '../../module/k8s';
+import * as k8sModels from '../../models';
+import {
+  ButtonBar,
+  history,
+  resourceObjPath,
+  ResourceDropdown,
+  ResourceIcon,
+} from '../utils';
+import { Checkbox } from '../checkbox';
+import { formatNamespacedRouteForResource } from '../../ui/ui-actions';
+
+const PvcDropdown = props => {
+  const kind = 'PersistentVolumeClaim';
+  const namespace = props.namespace;
+  const resources = [{ kind, namespace }];
+  const selectedKey = props.selectedKey;
+  const required = props.required;
+  return (
+    <ResourceDropdown
+      {...props}
+      desc="Storage"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select persistent volume claim"
+      selectedKey={selectedKey}
+      required={required}
+    />
+  );
+};
+
+// Helper functions for storage operations
+const createVolume = (name, claimName) => ({
+  name,
+  persistentVolumeClaim: {
+    claimName,
+  },
+});
+
+const createVolumeMount = (name, mountPath, subPath, readOnly) => {
+  const mount: any = {
+    name,
+    mountPath,
+    readOnly,
+  };
+
+  if (subPath) {
+    mount.subPath = subPath;
+  }
+
+  return mount;
+};
+
+// Gets the mount paths currently defined in the pod template.  An
+// optional filter function for matching specific containers can be used.
+const getMountPaths = (podTemplate, /* optional function */ matchContainer) => {
+  const mountPaths = [];
+  const containers = _.get(podTemplate, 'spec.containers', []);
+  _.each(containers, function(container) {
+    if (matchContainer && !matchContainer(container)) {
+      return;
+    }
+
+    const mounts = _.get(container, 'volumeMounts', []);
+    _.each(mounts, function(mount) {
+      mountPaths.push(mount.mountPath);
+    });
+  });
+
+  return mountPaths;
+};
+
+//Return kind type give the plural
+const getKindFromPlural = plural => {
+  const ko = _.find(k8sModels, model => model.plural === plural);
+  return ko.kind;
+};
+
+const AttachStorageForm = () =>
+  class StorageFormComponent extends React.Component<
+    StorageProps,
+    StorageState
+    > {
+    constructor(props) {
+      super(props);
+      const existingResource = _.pick(props.obj, ['metadata', 'spec']);
+      const resourceObj = _.defaultsDeep({}, props.fixed, existingResource, {
+        apiVersion: 'v1',
+        data: {},
+        kind: props.kind,
+        metadata: {
+          name: '',
+        },
+        spec: {},
+      });
+
+      this.state = {
+        resourceObj,
+        inProgress: false,
+        formData: {},
+        readOnly: false,
+        disableForm: false,
+        checkboxItems: [
+          {
+            value: 'Mount as read-only',
+          },
+        ],
+        selectedOptions: [],
+      };
+    }
+
+    componentDidMount() {
+      const { kind, metadata } = this.state.resourceObj;
+      const ko = _.find(k8sModels, model => model.plural === kind);
```

Comment:
`k8sModels` is not guaranteed to be initialized on mount. You need to use `connectToPlural`. Or if you want use `ResourceLink` I think you don't need it?